src/Microsoft.ML.OnnxTransformer/OnnxTypes.cs

```diff
@@ -0,0 +1,551 @@
+                {
+                    // Entering this scope means a ONNX tensor is found.
+
+                    var type = GetScalarDataViewType(typeProto.TensorType.ElemType).RawType;
+                    var methodInfo = typeof(NamedOnnxValue).GetMethod(nameof(NamedOnnxValue.AsTensor));
+                    var methodSpecialized = methodInfo.MakeGenericMethod(type);
+
+                    // NamedOnnxValue to Tensor.
+                    Func<NamedOnnxValue, object> caster = (NamedOnnxValue value) => methodSpecialized.Invoke(value, new object[] { });
+
+                    resultedType = typeof(Tensor<>).MakeGenericType(type);
+
+                    return caster;
+                }
+            }
+            else if (typeProto.ValueCase == OnnxCSharpToProtoWrapper.TypeProto.ValueOneofCase.SequenceType)
+            {
+                // Now, we see a Sequence in ONNX. If its element type is T, the variable produced by
+                // ONNXRuntime would be typed to IEnumerable<T>.
+
+                // Find a proper caster (a function which maps NamedOnnxValue to a .NET object) for the element in
+                // the ONNX sequence. Note that ONNX sequence is typed to IEnumerable<NamedOnnxValue>, so we need
+                // to convert NamedOnnxValue to a proper type such as IDictionary<>.
+                var elementCaster = GetDataViewValueCasterAndResultedType(typeProto.SequenceType.ElemType, out Type elementType);
+
+                // Set the .NET type which corresponds to the first input argument, typeProto.
+                resultedType = typeof(IEnumerable<>).MakeGenericType(elementType);
+
+                // Create the element's caster to map IEnumerable<NamedOnnxValue> produced by ONNXRuntime to
+                // IEnumerable<elementType>.
+                var methodInfo = typeof(CastHelper).GetMethod(nameof(CastHelper.CastOnnxSequenceToIEnumerable));
+                var methodSpecialized = methodInfo.MakeGenericMethod(typeof(NamedOnnxValue), elementType);
+
+                // Use element-level caster to create sequence caster.
+                Func<NamedOnnxValue, object> caster = (NamedOnnxValue value) =>
+                {
+                    var enumerable = value.AsEnumerable<NamedOnnxValue>();
+                    return methodSpecialized.Invoke(null, new object[] { enumerable, elementCaster });
+                };
+
+                return caster;
+            }
+            else if (typeProto.ValueCase == OnnxCSharpToProtoWrapper.TypeProto.ValueOneofCase.MapType)
+            {
+                // Entering this scope means a ONNX Map (equivalent to IDictionary<>) will be produced.
+
+                var keyType = GetNativeScalarType(typeProto.MapType.KeyType);
+                var valueType = GetNativeType(typeProto.MapType.ValueType);
+
+                // The resulted type of the object returned by the caster below.
+                resultedType = typeof(IDictionary<,>).MakeGenericType(keyType, valueType);
+
```

Comment:
Remove.

Reply:
We can't remove it because it's an `out` argument.

---
In reply to: [298418480](https://github.com/dotnet/machinelearning/pull/3881#discussion_r298418480) [](ancestors = 298418480)